### ISSUE TITLE
FormatException: Trying to read MMM from May 28 19:50 2023 at 0

### ISSUE THREAD

**Issue by rollodev** [salim-lachdhaf/dartFTP#17]
When I try to list the directory with:

```
final ftpConnect = FTPConnect(event.configFtp.address, user: event.configFtp.username, pass: event.configFtp.password, port: 21);
  ftpConnect.listCommand = ListCommand.LIST;
  await ftpConnect.connect();
  await ftpConnect.listDirectoryContentOnlyNames();
  ```

  I receive the following error:
  ```
  E/flutter ( 9458): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: FormatException: Trying to read MMM from May 28 19:50 2023 at 0
E/flutter ( 9458): #0      _DateFormatField.throwFormatException (package:intl/src/intl/date_format_field.dart:87:5)
E/flutter ( 9458): #1      _DateFormatPatternField.parseField (package:intl/src/intl/date_format_field.dart:350:7)
E/flutter ( 9458): #2      _DateFormatPatternField.parse (package:intl/src/intl/date_format_field.dart:261:5)
E/flutter ( 9458): #3      DateFormat._parse (package:intl/src/intl/date_format.dart:384:13)
E/flutter ( 9458): #4      DateFormat.parse (package:intl/src/intl/date_format.dart:315:7)
E/flutter ( 9458): #5      new FTPEntry._parseLIST (package:ftpconnect/src/ftp_entry.dart:198:40)
E/flutter ( 9458): #6      new FTPEntry._parseListCommand (package:ftpconnect/src/ftp_entry.dart:150:23)
E/flutter ( 9458): #7      new FTPEntry.parse (package:ftpconnect/src/ftp_entry.dart:56:23)
E/flutter ( 9458): #8      FTPDirectory.directoryContent.<anonymous closure> (package:ftpconnect/src/commands/directory.dart:86:20)
E/flutter ( 9458): #9      List.forEach (dart:core-patch/growable_array.dart:416:8)
E/flutter ( 9458): #10     FTPDirectory.directoryContent (package:ftpconnect/src/commands/directory.dart:83:59)
E/flutter ( 9458): <asynchronous suspension>
E/flutter ( 9458): #11     FTPDirectory.directoryContentNames (package:ftpconnect/src/commands/directory.dart:95:16)
E/flutter ( 9458): <asynchronous suspension>
E/flutter ( 9458): #12     FtpBloc._onFtpConnect (package:remanfredi/presentation/transport/bloc/ftp_bloc.dart:31:21)
E/flutter ( 9458): <asynchronous suspension>
E/flutter ( 9458): #13     Bloc.on.<anonymous closure>.handleEvent (package:bloc/src/bloc.dart:229:13)
E/flutter ( 9458): <asynchronous suspension>
E/flutter ( 9458):
  ```

**Comment by MarkKh792** [comment]
Hello, I got the same problem. This was caused by parsing a date string using `DateFormat`'s `parse()` method without specifying the locale:
`_modifyTime = DateFormat(format).parse(date);` (198th line of ftp_entry.dart).
The `DateFormat` class uses your device's default locale to parse the date string, so if your device is set to a locale other than English (in this case), it might throw a `FormatException`.
So I added the 'en_US' locale. The result is
`_modifyTime = DateFormat(format, 'en_US').parse(date);`
Hope this helps.
